### PR TITLE
Make minikube functional

### PR DESCRIPTION
**Pull request by OlafSzmidt**
**This PR will:**
- Ignore Mac generated files.
- Add a pip requirement for kubernetes.
- Patch the ingress to add the new game path for the appropriate service.
- Change the healthcheck to check for each game, not root.
- Restart the ingress ~~(if running)~~ and run it again. at each `./run.py -k`
- Add the ingress instance to be killed and restarted in `restart_pods`.
- Pin kubernetes version.
- Pass a separate path and host, and set the port to 0 always.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -76,3 +76,6 @@ aimmo-game/.hypothesis
 
 # Unity Build
 players/static/unity/**
+
+# Mac Generated Files
```
Comment (CelineBoudier) [inline .gitignore]: who's got a fancy new computer ;)?

File: aimmo-game-creator/worker_manager.py [hunk 2]
```diff
@@ -168,10 +169,18 @@ class KubernetesWorkerManager(WorkerManager):
 
     def __init__(self, *args, **kwargs):
         self._api = pykube.HTTPClient(pykube.KubeConfig.from_service_account())
+        kubernetes.config.load_incluster_config()
+        self._api_instance = kubernetes.client.ExtensionsV1beta1Api()
```
Comment (CelineBoudier) [inline aimmo-game-creator/worker_manager.py]: food for thought - is is stable? looks like they might rename that methid anytime - should we bind all the BetaThing named methods with a more generic name so it's easier to upgrade later?

File: aimmo-game-creator/worker_manager.py [hunk 3]
```diff
@@ -285,6 +308,7 @@ def create_worker(self, id, data):
             else:
                 raise
         self._create_game_rc(id, data)
+        self._add_path_to_ingress(id)
```
Comment (CelineBoudier) [inline aimmo-game-creator/worker_manager.py]: will we need to remove paths from ingress at some point?

File: aimmo-game-creator/worker_manager.py (hunk 2 above)
Comment (CelineBoudier) [reply aimmo-game-creator/worker_manager.py]: saw v1beta1 it is written in the ingress YAML file, but kubernetes is installed as latest version in the game creator. But Fixed version in minikube requirement...?

File: aimmo-game-creator/worker_manager.py (hunk 3 above)
Comment (OlafSzmidt) [reply aimmo-game-creator/worker_manager.py]: When we delete games, but we don't have that functionality yet

File: .gitignore (hunk 1 above)
Comment (OlafSzmidt) [reply .gitignore]: XD

File: aimmo-game-creator/worker_manager.py (hunk 3 above)
Comment (OlafSzmidt) [reply aimmo-game-creator/worker_manager.py]: 👍

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -25,3 +25,19 @@ The new starter / up-for-grabs issues are listed with the [help wanted label](ht
...
 * When a PR is accepted, **congrats!** It will be merged on master.
+
+## Some conventions to keep in mind...
+We follow PEP8 convention quite strictly however we do make a few exceptions to this rule. They are as follows:
+
+#### Maximum Line Length
+PEP8 specifies that the maximum line length is to be limited to *79 characters* with some lines even shorter than this. We replace this rule and try to stick to *90 characters*.
+
+#### Line Breaking after/before binary operators
+The internet seems be divided and PEP8 had some major changes when it comes to this rule. We have agreed that we want to strictly use binary operators on the **next** line, as opposed to before the line break.
```
Comment (CelineBoudier) [inline CONTRIBUTING.md]: The internet seems TO be divided

File: aimmo-game-creator/worker_manager.py (hunk 2 above)
Comment (OlafSzmidt) [reply aimmo-game-creator/worker_manager.py]: Discussed earlier today. Wrapper adds complexity to debugging and that's the official API name.

File: aimmo-game-creator/worker_manager.py [hunk 5]
```diff
@@ -268,8 +277,42 @@ def _create_game_service(self, id, _config):
...
+            }
+        ]
+
+        self._api_instance.patch_namespaced_ingress('aimmo-ingress', 'default', patch)
+
+    def _remove_path_from_ingress(self, game_id):
+        backend = kubernetes.client.V1beta1IngressBackend('game-%s' % game_id, 80)
+        path = kubernetes.client.V1beta1HTTPIngressPath(backend, '/game-%s' % game_id)
+        ingress = self._api_instance.list_namespaced_ingress("default").items[0]
+        paths = ingress.spec.rules[0].http.paths
```
Comment (CelineBoudier) [inline aimmo-game-creator/worker_manager.py]: is it safe to directly get [0] of things here?

Comment (OlafSzmidt) [reply aimmo-game-creator/worker_manager.py]: In fact, no. The next level down from rules will be each host - and order isn't static on the ingress yaml files it seems :/

BUT..... I guess we can cover this by the tests we will write asap.

**Pull request merged**